packages/steiger-plugin-fsd/src/_lib/index-source-files.ts

```diff
@@ -28,6 +28,11 @@ export function indexSourceFiles(root: Folder): Record<string, SourceFile> {
   }
 
   for (const [layerName, layer] of Object.entries(getLayers(root))) {
+    // Even though files that are directly inside a layer are not encouraged by the FSD and are forbidden in most cases
+    // (except for an index/root file for the app layer as an entry point to the application), users can still add them.
+    // So, we need to index all files directly inside a layer to find errors.
+    walk(layer, { layerName: layerName as LayerName, sliceName: null, segmentName: null })
```

Comment:
@illright 
Actually, you can postpone your review. I will fix the problem with this line. Because I just realized that it will go through all nested files/directories. And then we walk through them again in the lines below